driver-core/src/main/java/com/datastax/driver/core/AbstractTableMetadata.java

```diff
@@ -258,14 +258,19 @@ protected StringBuilder appendOptions(StringBuilder sb, boolean formatted) {
     sb.append("WITH ");
     if (options.isCompactStorage()) and(sb.append("COMPACT STORAGE"), formatted);
     if (!clusteringOrder.isEmpty()) and(appendClusteringOrder(sb), formatted);
-    sb.append("read_repair_chance = ").append(options.getReadRepairChance());
-    and(sb, formatted)
-        .append("dclocal_read_repair_chance = ")
-        .append(options.getLocalReadRepairChance());
+    if (cassandraVersion.getMajor() < 4)
+      sb.append("read_repair_chance = ").append(options.getReadRepairChance());
+    else sb.append("read_repair = 'BLOCKING'");
+    if (cassandraVersion.getMajor() < 4)
+      and(sb, formatted)
+          .append("dclocal_read_repair_chance = ")
+          .append(options.getLocalReadRepairChance());
```

Comment:
This was done to get tests passing against C* 4.0. I don't think this is the right way to address this as the changed table options are just hard-coded here, with no way to change them or retrieve them in a TableMetadataOptions object.

Reply:
created https://datastax-oss.atlassian.net/browse/JAVA-2730 for this

Reply:
We also need to amend the code that parses the options. That's done in the constructor of `TableOptionsMetadata`.

As far as outputting the options conditionally, I think that's the right way to do it. That's how it was already handled previously (driver 3 hard-codes a lot more things).